dll/win32/shell32/dialogs/folder_options.cpp

```diff
@@ -1443,20 +1584,55 @@ InsertFileType(HWND hDlgCtrl, WCHAR * szName, PINT iItem, WCHAR * szFile)
             RegQueryValueExW(hKey, L"EditFlags", NULL, NULL, (LPBYTE)&Entry->EditFlags, &dwSize);
     }
 
+    /* convert extension to upper case */
+    wcscpy(Entry->FileExtension, szName);
+    _wcsupr(Entry->FileExtension);
+
+    /* get icon */
+    if (!GetFileTypeIconsByKey(hKey, Entry))
+    {
+        // set default icon
+        const INT iIndex = IDI_SHELL_FOLDER_OPTIONS;
+        Entry->hIconLarge = LoadIconW(shell32_hInstance, MAKEINTRESOURCEW(iIndex));
```

Comment:
Please put the value of iIndex inside the following two function calls: this will avoid the unnecessary creation of temporary variables.

Reply:
OK.